spec/rubocop/cop/layout/closing_heredoc_indentation_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Layout::ClosingHeredocIndentation do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) do
+    RuboCop::Config.new('Layout/ClosingHeredocIndentation' => cop_config)
+  end
+  let(:cop_config) { { 'Enabled' => true } }
+
+  it 'accepts correctly indented closing heredoc' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      class Test
+        def foo
+          <<-SQL
+            bar
+          SQL
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offence for bad indentation of a closing heredoc' do
+    expect_offense(<<-RUBY.strip_indent)
+      class Test
+        def foo
+          <<-SQL
+            bar
+        SQL
+      ^^^^^ `SQL` is not aligned with `<<-SQL`.
```

Comment:
Can we get rid of the overshooting highlight?

Reply:
I'd love to! Might need a pointer though. I _think_ the root is that https://github.com/bbatsov/rubocop/blob/ef1daf7e97e2362ceaac2a933b243a753d1bd3a9/lib/rubocop/cop/offense.rb#L147-L149 is returning `0...5`, potentially related to https://github.com/bbatsov/rubocop/blob/ef1daf7e97e2362ceaac2a933b243a753d1bd3a9/lib/rubocop/cop/offense.rb#L113-L115 returning 0.

I think it's related to
```ruby
irb(main):009:0> Parser::CurrentRuby.parse("     <<-SQL\nfoo\n    SQL").loc
=> #<Parser::Source::Map::Heredoc:0x0000555e4ccccf68 @heredoc_body=#<Parser::Source::Range (string) 12...16>, @heredoc_end=#<Parser::Source::Range (string) 16...23>, @expression=#<Parser::Source::Range (string) 5...11>, @node=s(:str, "foo\n")>
```
Would we expect `heredoc_end` to range from `20...23`?